student/main.go

```diff
@@ -0,0 +1,118 @@
+package main
+
+import (
+	"fmt"
+	"reflect"
+)
+
+type Student struct {
+	name  string
+	age   int
+	grade int
+}
+
+func createStudent() Student {
+	var name string
+	var age int
+	var grade int
+	fmt.Print("Enter name: ")
+	fmt.Scan(&name)
+	fmt.Print("Enter age: ")
+	fmt.Scan(&age)
+	fmt.Print("Enter grade: ")
+	fmt.Scan(&grade)
+	if reflect.TypeOf(name).Kind() != reflect.String {
+		fmt.Println("Invalid name")
+		return Student{"", 0, 0}
+	}
+	if reflect.TypeOf(age).Kind() != reflect.Int {
+		fmt.Println("Invalid age")
+		return Student{"", 0, 0}
+	}
+	if reflect.TypeOf(grade).Kind() != reflect.Int {
+		fmt.Println("Invalid grade")
+		return Student{"", 0, 0}
+	}
+	return Student{name, age, grade}
+}
+
+func (s Student) find (students []Student, name string) Student {
+	for _, student := range students {
+		if student.name == name {
+			return student
+		}
+	}
+	return Student{"", 0, 0}
+}
+func main() {
+	students := make([]Student, 0)
+	students = append(students, Student{"John", 21, 90})
+	students = append(students, Student{"Jane", 21, 95})
+	students = append(students, Student{"Bob", 21, 92})
+	for chosen := true; chosen; {
+		fmt.Println("1. Print all students")
```

Comment:
Burdaki gibi arama tiplerine sahip olacağında bunları iota kullanarak constant olarak tutabilirsin.
ref: https://golangbyexample.com/iota-in-golang/